### PR TITLE
[#49] [Android] [UI] As a user, I can see multiple choices answers

### LINKED ISSUE #49: [Android] [UI] As a user, I can see multiple choices answers
# Why

User can select multiple answers for a question has `pick` is `any`

# Acceptance Criteria

- Display a list of answers as a multichoice list based when question's `display_type` is `choice` and `pick` is `any`

Image

[Figma](https://www.figma.com/file/GjRPOjDyZ6f4EDL3wKarRK/Challenge---Mobile-App?node-id=34-1172&t=1xiBQA7GNv0kEx7j-0)

### PR DESCRIPTION
**Pull request by AVI5HEK**
- Close #49 

## What happened 👀

Provide a description of the **changes** this pull request brings to the codebase. Additionally, when the pull request is still being worked on, a checklist of the planned changes is welcome to track progress.

## Insight 📝

Describe in detail why this solution is the most appropriate, which solution you tried but did not go with, and how to test the changes. References to relevant documentation are welcome as well.

## Proof Of Work 📹

image

### CODE REVIEW

File: androidApp/src/main/java/co/nimblehq/avishek/phong/kmmic/android/ui/screen/surveydetail/TextFields.kt [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.tooling.preview.PreviewParameter
+import androidx.compose.ui.unit.dp
+import co.nimblehq.avishek.phong.kmmic.android.ui.common.PrimaryTextField
+import co.nimblehq.avishek.phong.kmmic.presentation.uimodel.AnswerUiModel
+
+@Composable
+fun TextFields(
```
Comment (suho) [inline androidApp/src/main/java/co/nimblehq/avishek/phong/kmmic/android/ui/screen/surveydetail/TextFields.kt]: Same with this one, please check conflict

**Review by suho — APPROVED** [review]

File: androidApp/src/main/java/co/nimblehq/avishek/phong/kmmic/android/ui/screen/surveydetail/MultiChoiceForm.kt [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+import co.nimblehq.avishek.phong.kmmic.presentation.uimodel.SurveyAnswerUiModel
+
+@Composable
+fun MultiChoiceForm(
+    surveyAnswerUiModels: List<SurveyAnswerUiModel>,
+    modifier: Modifier = Modifier,
+) {
+    var checkedAnswers by remember { mutableStateOf(emptySet<SurveyAnswerUiModel>()) }
+    LazyColumn(modifier = modifier.padding(horizontal = 80.dp)) {
+        items(surveyAnswerUiModels.size) { index ->
```
Comment (kaungkhantsoe) [inline androidApp/src/main/java/co/nimblehq/avishek/phong/kmmic/android/ui/screen/surveydetail/MultiChoiceForm.kt]: It is a good practice to include `key` if you use `LazyColumn` to reduce the `recomposition` count.
```suggestion
        items(
            count = surveyAnswerUiModels.size,
            key = { index -> surveyAnswerUiModels[index].id },
        ) { index ->
```

File: androidApp/src/main/java/co/nimblehq/avishek/phong/kmmic/android/ui/screen/surveydetail/MultiChoiceForm.kt [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+}
+
+@Preview
+@Composable
+fun MultipleChoiceFormPreview(
+    @PreviewParameter(SurveyDetailScreenPreviewParameterProvider::class)
+    params: SurveyDetailScreenPreviewParameterProvider.Params,
+) {
+    MultiChoiceForm(
+        surveyAnswerUiModels = params.survey.questionUiModels[0].answers
```
Comment (kaungkhantsoe) [inline androidApp/src/main/java/co/nimblehq/avishek/phong/kmmic/android/ui/screen/surveydetail/MultiChoiceForm.kt]: This too shows a lot of unnecessary Previews. Please refactor a bit to reduce the number of previews for this composable.

Comment (AVI5HEK) [reply androidApp/src/main/java/co/nimblehq/avishek/phong/kmmic/android/ui/screen/surveydetail/MultiChoiceForm.kt]: I'll update all the previews of this PR and others onward. 🙇🏽

File: androidApp/src/main/java/co/nimblehq/avishek/phong/kmmic/android/ui/screen/surveydetail/MultiChoiceForm.kt (hunk 2 above)
Comment (AVI5HEK) [reply androidApp/src/main/java/co/nimblehq/avishek/phong/kmmic/android/ui/screen/surveydetail/MultiChoiceForm.kt]: Updated here https://github.com/nimblehq/avishek-phong-kmm-ic/pull/125/commits/56152ad459b92c5a8576eb202c4f690396a43267

File: androidApp/src/main/java/co/nimblehq/avishek/phong/kmmic/android/ui/screen/surveydetail/MultiChoiceForm.kt (hunk 3 above)
Comment (AVI5HEK) [reply androidApp/src/main/java/co/nimblehq/avishek/phong/kmmic/android/ui/screen/surveydetail/MultiChoiceForm.kt]: Fixed: https://github.com/nimblehq/avishek-phong-kmm-ic/pull/125/commits/c5c80968d0715422a83b2ba7adce5fc21a12ad7d

**Review by kaungkhantsoe — APPROVED** [review]

**Pull request merged**